packages/flutter/lib/src/widgets/focus_manager.dart

```diff
@@ -738,6 +788,9 @@ class FocusNode with DiagnosticableTreeMixin, ChangeNotifier {
 
   // Note that this is overridden in FocusScopeNode.
   void _doRequestFocus() {
+    if (!canRequestFocus) {
```

Comment:
I guess it will be ok as the node is basically the same as skip-traverse and cannot be explicit focus, but it still need to be in the tree to pass on focus to its parent in case of removal.

Reply:
Yes, exactly.